src/Providers/NorthwesternUiServiceProvider.php

```diff
@@ -44,6 +44,21 @@ public function boot()
                 'dsn' => config('northwestern-theme.sentry-dsn'),
                 'environment' => config('app.env'),
                 'tunnel' => Route::has('sentry.tunnel') ? route('sentry.tunnel', [], false) : null,
+                'integrations' => [], // add to integrations in blade since it has to init js object
+                'enable_apm' => (bool)config('northwestern-theme.sentry-enable-apm-js'),
+                'tracesSampleRate' => config('northwestern-theme.sentry-traces-sample-rate'),
+            ]);
+
+            $user = auth()->user();
+
+            if (! $user) {
+                return;
+            }
+
+            $view->with('user', [
```

Comment:
This would probably need to use a view composer so the variable is scoped to `purple-chrome` -- otherwise any consuming application that has `$user` as a variable for a page's view will interfere.

The data listed here is also specific to SOAP. This package is used by several applications, so it cannot make any assumptions about fields available on `User`.

In SOAP, there's a `AddSentryContext` middleware that sets some user data. Can this can be refactored into something this package provides, and then both the backend & frontend SDKs share the same user context?

Reply:
I made a `UserContextGetter` class for shared user context in the frontend and backend. The `segment` property value is now optional depending on whether the user's class has such a method - since both the frontend and backend need this property and the `segment` method only exists in SOAP, I was thinking this approach may be more flexible.

If it looks good, I'll open a PR to implement the class in SOAP.